apps/ftu/js/ui.js

```diff
@@ -123,6 +123,26 @@ var UIManager = {
       this[toCamelCase(name)] = document.getElementById(name);
     }.bind(this));
 
+    // Setup settings observers
+    this._settingsObserveHandler = {
+      'geolocation.enabled': {
+        defaultValue: this.geolocationCheckbox.checked,
+        callback: function(value) {
+          if (this.geolocationCheckbox.checked != value) {
```

Comment:
I actually didn't do the strict equality check on purpose as I'm not sure the callback will give me a value that I can match exactly against .checked (e.g null !== false), but I assume it will cast/coerce to true/false appropriately. Maybe it would be more explicit to do that on value before testing it against .checked?